### DISCUSSION TITLE
1.6.1

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#747] [Обновления]
## Список изменений

- Теперь, после инициализации кнопка скрывается, а не ждет пока вы куда-нибудь нажмете/наведетесь на видео
- В качестве эксперимента, была убрана регистрация видео, которые изначально были замучены (такие видео могли использоваться для заднего фона или как вспомогательные элементы для основного плеера)
- Исправлена ошибка из-за которой на сайтах без поддержки субтитров кнопка могла не показываться
- Исправлена ошибка из-за которой кнопка не исчезала на Pornhub (#700)
- Ускорена работа VideoObserver (#740)
- Временно, поддержка `www.reddit.com` заменена на `old.reddit.com` из-за багов с инициализацией
- Добавлена поддержка большего числа инстансов Peertube
- Добавлено проксирование субтитров (#739)
- Добавлена поддержка Poketube (#726)
- Исправлена работа Peertube
- Исправлена работа Dailymotion ([#728#discussioncomment-10300700](https://github.com/ilyhalight/voice-over-translation/discussions/728#discussioncomment-10300700))
- Исправлена работа Google Drive (#737)

## What's Changed
* Доработка VideoObserver by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/740

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.6.0...1.6.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.6.1'>1.6.1</a>.</em>

**Comment by jaroshevskii** [comment]
Why is there no Cloudflare version?

**Reply by ilyhalight** [reply]
> Why is there no Cloudflare version?

Now, the regular version includes features from the cloudflare version

**Comment by FiLinX** [comment]
я вот чего то никак немогу сообразить, как же мне заставить VOT чтобы в его настройках, на язык оригинала - всегда дефолтно стояло Авто, а не энглишь ?
ну, я уже устал от того, что на многих роликах после нажатия Перевести видео и вроде как адекватного получения перевода (панелька высвечивает Отключить)  -  я при воспроизведении не слышу никакого перевода, лишь громкость оригинала приглушается до выставленного уровня в настройках..
и приходится блин каждый раз, прежде чем нажимать на Перевести видео, сначала заходить в троеточие и изменять там энглишь на авто для оригинала  -  и вот тогда только ролики идут с переводом (хотя язык то у тех роликов и был английский)
вобщем странности какие то..
это было ещё и на 1.6.0

**Reply by ilyhalight** [reply]
> я вот чего то никак немогу сообразить, как же мне заставить VOT чтобы в его настройках, на язык оригинала - всегда дефолтно стояло Авто, а не энглишь ?

В коде расширения поменяй `translateFromLang = "en"` на `translateFromLang =  "auto"` и убедись, что в функции getVideoData (7907 строка) для нужного сайта не устанавливается detectedLanguage. Тогда для всех сайтов по умолчанию будет ставиться авто

> ну, я уже устал от того, что на многих роликах после нажатия Перевести видео и вроде как адекватного получения перевода (панелька высвечивает Отключить) - я при воспроизведении не слышу никакого перевода, лишь громкость оригинала приглушается до выставленного уровня в настройках..

Это где такое происходит?

**Reply by FiLinX** [reply]
ну как где - на ютубе)  из десяти открытых в фоне буржуйских ютубин, обязательно хоть одна, но будет с таким глюком - типа сработавшее получение перевода, но по факту лишь приглушённый оригинал и без перевода..
перезагрузка (даже несколько раз) вкладки и нажатие Перевести видео - ничего не изменяет.. лечится только предварительным выставлением на Авто, и только после нажатием на Перевести видео..

кстати, заметил на нескольких таких несработавших  -  надпись Перевести видео изменялась на Отключить сразу (ну как из кешированных уже), но вот после того как обнаруживаю отсутствие звука перевода, перезагружаю вкладку, выставляю Авто, жму Перевести - и мне выводит обождите около минуты 
ну, то есть когда на оригинал выставлено Энглишь - и типо срабатывает получение перевода с кешированных  -  по факту то перевода то ещё никакого и небыло с энглиша закешировано  -  язык у роликов точно энглишь, без примесей
ну и сами ролики - они из моей ленты подписок, и некоторые вообще лишь по нескольку минут назад вышедшие

хотя. может быть это у меня как то в темпермонке иль ещё где хвосты настроек от предыдущих версий VOTа подвисли, я сейчас несколько версий чисто через меню темпермонки обновлял..
просто ведь я помню была ситуация с одним роликом, и я обратился с этим сюда к вам, где вы как раз подсказали мне на том ролике сменить Авто на Энглишь  -  и оно помогло!  но вот видать после этого этот Энглишь так и завис дефолтным..
хм.. попробую щас отключить и удалить 1.6.1 в темпермонке, перезагружу браузер на всякий, и поставлю VOTа вновь)

**Reply by FiLinX** [reply]
хм.. пока не разобрался до конца, потому как после переустановки скрипта пока не попадаются с подобным глюком..
но зато выяснил - что предварительная смена Энглишь на Авто, при запросе - всегда инициализирует ожидание нового перевода!!  хотя секунды назад ролик с дефолно стоящей Энглишь и мгновенно полученному переводу - шли в переводе)
короче Авто - принуждение к новой озвучке! о как.. не знал)

**Reply by ilyhalight** [reply]
> редварительная смена Энглишь на Авто, при запросе - всегда инициализирует ожидание нового перевода!! хотя секунды назад ролик с дефолно стоящей Энглишь и мгновенно полученному переводу - шли в переводе) короче Авто - принуждение к новой озвучке! о как.. не знал)

Так с любой языковой парой работает, будь то смена языка с которого переводится или на который переводится

**Comment by BAGUVIX90** [comment]
> Временно, поддержка www.reddit.com заменена на old.reddit.com из-за багов с инициализацией

Напомню что есть ещё new.reddit.com коем я обычно и пользуюсь. Что по поводу данного домена?

**Reply by ilyhalight** [reply]
> > Временно, поддержка [www.reddit.com](http://www.reddit.com) заменена на old.reddit.com из-за багов с инициализацией
> 
> Напомню что есть ещё new.reddit.com коем я обычно и пользуюсь. Что по поводу данного домена?

В планах есть вернуть и reddit.com и new.reddit.com, но, пока, не совсем понимаю как на них нормально отслеживать появление элемента с видео, чтобы оно стабильно работало

**Reply by BAGUVIX90** [reply]
Тут к сожалению я вам не помошник. Может в каком то специализированном сабреддите вопрос задать?

**Comment by ruSauron** [comment]
Просто спасибо за расширение. После того, как стали портить ютуб столкнулся с проблемой, что яндекс браузер на роликах ютуба довольно часто просто вис в случайные моменты, обычно в самом начале видео, при этом сильно нагружая процессор. Вероятно, что-то сломали, когда впихивали рекомендации по поиску видео на "правильных" платформах. С этим расширением вновь можно нормально пользоваться переводом видео.

**Comment by ShadowDick177** [comment]
Скачал расширение, но на видео не появляется подсказки для перевода

**Reply by ilyhalight** [reply]
Вы включили режим разработчика в настройках браузера?

**Reply by ShadowDick177** [reply]
тупанул, спасибо

**Comment by FiLinX** [comment]
скажите, я помойму уже давно когда то видел тут упоминания про https://rumble.com/videos?sort=views&date=today
но щас попал случайно туда, а панельки не вижу.. его не добавляли чтоль да?

**Reply by ilyhalight** [reply]
у меня работает, может быть у вас какая-то ссылка у видео, которая еще не добавлена

![firefox_2qAISpcLrb](https://github.com/user-attachments/assets/2c71f63b-379e-40f6-9a60-6578c58ee11a)

**Reply by FiLinX** [reply]
да я как бы вот с вышеданной ссылки на нью-тренды переходил смотрел..
например у меня там щас это выдаёт - https://i.imgur.com/0bh20yV.jpeg
и первые три ссылки такие:
https://rumble.com/v5da7ro-ep.-3439b-comms-blackout-has-begun-musk-releases-the-secret-weapon-the-plan.html
https://rumble.com/v5d8jv9-ep.-226-elon-and-x-bannes-in-brazil-trump-superseding-indictment-kamala-int.html
https://rumble.com/v5d9gl1-die-deutschen-wollen-es-offenbar-nicht-anders-weiter-so-als-groes-signal-de.html
и на их страницах панельки вообще нету((
нету даже и на эмбедках 
в консоли вроде тоже ничего про VOT нету  -  https://i.imgur.com/Vv6i5WJ.jpeg
но в тоже время скрипт из темпермонки вроде как задействуется - https://i.imgur.com/aTMVZeA.jpeg
или же не задействуется? потому как снизу там про эррор какой то с указанием на темпермонку - https://i.imgur.com/nDLoF89.jpeg
хотя не, эта эррор присутствует и на ютубинах, но там то панелька есть и всё работает

странно.. 
вин10, хромой-118, темпермонка-5.1.1, и VOT-1.6.1

попробовал щас на темпермонке включил галочку для срабатывания в инкогнито режиме  -  но и так тоже нету панельки там

**Reply by ilyhalight** [reply]
> да я как бы вот с вышеданной ссылки на нью-тренды переходил смотрел.. например у меня там щас это выдаёт - https://i.imgur.com/0bh20yV.jpeg и первые три ссылки такие: https://rumble.com/v5da7ro-ep.-3439b-comms-blackout-has-begun-musk-releases-the-secret-weapon-the-plan.html https://rumble.com/v5d8jv9-ep.-226-elon-and-x-bannes-in-brazil-trump-superseding-indictment-kamala-int.html https://rumble.com/v5d9gl1-die-deutschen-wollen-es-offenbar-nicht-anders-weiter-so-als-groes-signal-de.html и на их страницах панельки вообще нету(( нету даже и на эмбедках в консоли вроде тоже ничего про VOT нету - https://i.imgur.com/Vv6i5WJ.jpeg но в тоже время скрипт из темпермонки вроде как задействуется - https://i.imgur.com/aTMVZeA.jpeg или же не задействуется? потому как снизу там про эррор какой то с указанием на темпермонку - https://i.imgur.com/nDLoF89.jpeg хотя не, эта эррор присутствует и на ютубинах, но там то панелька есть и всё работает
> 
> странно.. вин10, хромой-118, темпермонка-5.1.1, и VOT-1.6.1
> 
> попробовал щас на темпермонке включил галочку для срабатывания в инкогнито режиме - но и так тоже нету панельки там

А в списке юзерскриптов тамперманки показывается расширение? Если да, то попробуйте вручную обновить расширение до самой последней версии через кнопку "установить скрипт" на главной репозитория (будет так же показываться 1.6.1, но должно заработать)

должно выглядеть примерно так:
![firefox_gbaQlRjUva](https://github.com/user-attachments/assets/3f9561f6-1c77-427c-9c76-51d09bce29b6)


> потому как снизу там про эррор какой то с указанием на темпермонку - https://i.imgur.com/nDLoF89.jpeg
хотя не, эта эррор присутствует и на ютубинах, но там то панелька есть и всё работает

это ошибка из-за того, что не удается найти карту кода для одной из библиотек, ничего в этом критичного нет

**Reply by FiLinX** [reply]
ыыыыы)))  вот это я вовремя затеял..  переустановить то переустановил, но пробую щас открыть любую рамблу - а мне в лоб:
Because of French government demands to remove creators from our platform - https://i.imgur.com/DE7R5dO.jpeg
)))))))  жжесть...

но по теме,  -  а как же это так то, я прошлую 1.6.1 устанавливал обновлением в самой темпермонке поверх 1.6.0 ещё 14 дней назад
так вот сейчас когда кликнул с главной и открылась страница темпермонки с Переустановить, то там в поле Изменения - мне выдало вот это: Не удаётся отобразить этот скрипт, слишком большие различия
гы)  
линк с релизов - https://github.com/ilyhalight/voice-over-translation/releases/download/1.6.1/vot.user.js
и с главной, с Установите скрипт - https://raw.githubusercontent.com/ilyhalight/voice-over-translation/master/dist/vot.user.js
и внатуре, щас выкачал по обеим ссылкам - и файлы то довольно сильно различаются)
хотя в обоих - Обновлён | 19.08.2024, 18:03:30
было - https://i.imgur.com/rII3y3x.jpeg          и сейчас - https://i.imgur.com/zFkMM4m.jpeg
хех)

но чё, всё оказалось впустую  -  хранцузы "демократизировали" всё по самое небалуйся уже))
и не зря как видно основатель Рамблы стремался и свалил из Франции как только с Дуровым всё произошло..

**Reply by ilyhalight** [reply]
> - а как же это так то, я прошлую 1.6.1 устанавливал обновлением в самой темпермонке поверх 1.6.0 ещё 14 дней назад
> так вот сейчас когда кликнул с главной и открылась страница темпермонки с Переустановить, то там в поле Изменения - мне выдало вот это: Не удаётся отобразить этот скрипт, слишком большие различия

Так показывает потому, что еще с релиза 1.5.1 вернулся к изначальной концепции - релизы публикуются раз в N время как отдельная версия [тут](https://github.com/ilyhalight/voice-over-translation/releases), а на главной всегда доступна самая актуальная версия, даже, если она еще в "разработке" (за исключением, если версия содержит какие-то глобальные изменения, которые нужно дополнительно тестировать). Как бы получается и бета, но по стабильности остается +-такая же, как и релизные версии

*версия не меняется, чтобы тамперманки и другие подобные расширения не автообновлялись до нее раньше времени

> хотя в обоих - Обновлён | 19.08.2024, 18:03:30

Скорее всего, тамперманки как-то неправильно информацию подхватывает

**Reply by FiLinX** [reply]
урряя!  сегодня рамбла вновь работает  -  и панелька при обновлённом 1.6.1 на месте!! 
Спасибо!
